api/storage_service.cc

```diff
@@ -694,29 +694,21 @@ void set_storage_service(http_context& ctx, routes& r, sharded<service::storage_
         auto keyspace = validate_keyspace(ctx, req->param);
         auto table_infos = parse_table_infos(keyspace, ctx, req->query_parameters, "cf");
         apilog.info("force_keyspace_cleanup: keyspace={} tables={}", keyspace, table_infos);
+        std::exception_ptr ex = nullptr;
         if (!co_await ss.local().is_cleanup_allowed(keyspace)) {
             auto msg = "Can not perform cleanup operation when topology changes";
             apilog.warn("force_keyspace_cleanup: keyspace={} tables={}: {}", keyspace, table_infos, msg);
-            co_await coroutine::return_exception(std::runtime_error(msg));
+            ex = std::make_exception_ptr(std::runtime_error(msg));
```

Comment:
Why?  What's the reason to go ahead and create the task, that is going to fail with this error?

I don't think this is needed for the synchronous interface.
I also doubt if we would do that over an async interface.
It's ok to return synchronous errors in either case.
For example, one of the parameters could be wrong (missing keyspace / table).

Reply:
I thought it would be more consistent if user could see the task concerning the operation he started even if it fails immediately.

The user gets the exception from `force_keyspace_compaction` though, so I agree. Will change.